model-optimizer/mo/utils/ir_reader/extenders/ctc_greedy_decoder_extender.py

```diff
@@ -0,0 +1,31 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from extensions.front.tf.CTCGreedyDecoder import CTCGreedyDecoderReplacement
+from extensions.ops.ctc_greedy_decoder import CTCGreedyDecoderOp
+from mo.utils.graph import Node
+from mo.utils.ir_reader.extender import Extender
+
+
+class CTCGreedyDecoder_extender(Extender):
+    op = 'CTCGreedyDecoder'
+
+    @staticmethod
+    def extend(op: Node):
+        if op.graph.graph['cmd_params'].framework == 'tf':
```

Comment:
Extender is always work with OpenVINO representation of the operations. So there should be no any condition like this. The shape infer function should work according to the operation specification.